### PR TITLE
Valid W3C URL in fsi_file_path and fsi_file_asset

### LINKED ISSUE #43: fsi_file_path - invalid generated url's
url's returned by `fsi_file_path()` should return W3C walid url, so each part of path should be passed through `url_encode`

### PR DESCRIPTION
**Pull request by bartekgruszka**
Each part of an URL is encoded by `rawurlencode()` to be W3C valid.
Updated test suites to validate encoded URLs.
Closes #43

### CODE REVIEW

**Comment by JarekW** [comment]
:+1:

**Comment by rn0** [comment]
:+1:

**Comment by chives** [comment]
the same encoding should also be applied in this method https://github.com/fsi-open/doctrine-extensions-bundle/blob/master/Resolver/FSiFilePathResolver.php#L61

**Comment by bartekgruszka** [comment]
@chives done

File: spec/FSi/Bundle/DoctrineExtensionsBundle/Resolver/FSiFilePathResolverSpec.php [hunk 1]
```diff
@@ -111,7 +117,7 @@ function it_generate_url_with_passed_prefix_even_if_there_is_a_prefix_in_globals
      */
     function it_generate_fsi_file_basename($file)
     {
-        $file->getName()->shouldBeCalled()->willReturn('TestFolder/File/file.jpg');
+        $file->getName()->willReturn('TestFolder/File/file.jpg');
 
         $this->fileBasename($file)->shouldReturn('file.jpg');
```
Comment (chives) [inline spec/FSi/Bundle/DoctrineExtensionsBundle/Resolver/FSiFilePathResolverSpec.php]: maybe we should test if the filename is encoded here like [here](https://github.com/fsi-open/doctrine-extensions-bundle/pull/45/files#diff-ae6ddbd14596047c574825c79155c0bdR50)

**Comment by chives** [comment]
:+1:

**Pull request merged**